### PR TITLE
[DO NOT MERGE UNTIL DEV-PREVIEW for Dart API category released] feat(flutter): update API for dev preview

### PR DESCRIPTION
**Pull request by ragingsquirrel3**
DO NOT MERGE UNTIL DEV-PREVIEW for Dart API category released.

This PR has updates for dev preview of amplify-flutter API category (implemented in Dart for wider platform support). Most of the changes are in REST API but there are a few additions for GraphQL as some missing features are added here.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/fragments/lib/graphqlapi/flutter/authz/30_multi.mdx [hunk 1]
```diff
@@ -1,15 +1,67 @@
...
+<Block name="Stable (Mobile)">
+
+  When you have configured multiple APIs, you can specify the name of the API as a parameter as the target for an operation:
+
+  ```dart
+  Future<void> main() async {
+    final operation = Amplify.API.mutate<String>(
+      request: GraphQLRequest(
+          document: graphQLDocumentString,
+          apiName: '[FRIENDLY-NAME-API-WITH-API-KEY]',
```
Comment (abdallahshaban557) [inline src/fragments/lib/graphqlapi/flutter/authz/30_multi.mdx:12]: can we clarify where friendly name comes from please?

**Review by abdallahshaban557 — COMMENTED** [review]
Left one comment. thank you for clarifying the piece on using customer graphQL requests!

**Review by abdallahshaban557 — APPROVED** [review]
LGTM!

File: src/fragments/lib-v1/restapi/flutter/getting-started/40_postTodo.mdx [hunk 2]
```diff
@@ -1,20 +1,48 @@
...
-}
-```
+  ```
+
+</Block>
+
+<Block name = "Developer Preview (Mobile, Web & Desktop)">
+
+  ```dart
+  Future<void> onTestApi() async {
```
Comment (Equartey) [inline src/fragments/lib-v1/restapi/flutter/getting-started/40_postTodo.mdx:31]: nit: I know this was the name before, but the other examples are all relevant to the operation. Could rename to `postTodo()`

**Review by Equartey — DISMISSED** [review]
Wonderful improvements! Just one nit.

File: src/fragments/lib-v1/restapi/flutter/getting-started/40_postTodo.mdx (hunk 2 above)
Comment (ragingsquirrel3) [reply src/fragments/lib-v1/restapi/flutter/getting-started/40_postTodo.mdx]: good suggestion, just updated

File: src/fragments/lib/restapi/native_common/getting-started/common.mdx [hunk 3]
```diff
@@ -61,7 +61,7 @@ import flutter11 from "/src/fragments/lib/restapi/flutter/getting-started/30_ini
 
 ## Make a POST Request
 
-Copy and paste the following code in your application so that it runs only once when the app starts:
+Send a POST request with a JSON body.
```
Comment (ragingsquirrel3) [inline src/fragments/lib/restapi/native_common/getting-started/common.mdx:64]: callout: This part applies to android and iOS as well but I thought it made more sense. LMK if anyone disagrees and I can change back.

**Review by abdallahshaban557 — APPROVED** [review]
LGTM!

File: src/fragments/lib-v1/restapi/flutter/fetch.mdx [hunk 4]
```diff
@@ -53,19 +77,47 @@ app.get('/todo', function(req, res) {
...
+
+</Block>
+
+<Block name="Developer Preview (Mobile, Web & Desktop)">
+
+  ```dart
+  Future<void> getTodo() async {
+    try {
+      final restOperation = Amplify.API.get(
+        'todo',
```
Comment (salihgueler) [inline src/fragments/lib-v1/restapi/flutter/fetch.mdx:110]: What was the reason of droppong the / in the beginning at all places?

File: src/fragments/lib-v1/restapi/native_common/getting-started/common.mdx [hunk 5]
```diff
@@ -61,7 +61,7 @@ import flutter11 from "/src/fragments/lib-v1/restapi/flutter/getting-started/30_
 
 ## Make a POST Request
 
-Copy and paste the following code in your application so that it runs only once when the app starts:
+Send a POST request with a JSON body.
```
Comment (salihgueler) [inline src/fragments/lib-v1/restapi/native_common/getting-started/common.mdx:64]: This affects all the platforms, is it designed to be that way?

**Review by salihgueler — APPROVED** [review]

File: src/fragments/lib-v1/restapi/native_common/getting-started/common.mdx (hunk 5 above)
Comment (ragingsquirrel3) [reply src/fragments/lib-v1/restapi/native_common/getting-started/common.mdx:64]: Yes, left a callout comment about this but I found it confusing and thought maybe it's a mistake, so changed it here. (Not sure why someone would execute a REST API call only when the app starts). If I'm wrong happy to change back.

File: src/fragments/lib-v1/restapi/flutter/fetch.mdx (hunk 4 above)
Comment (ragingsquirrel3) [reply src/fragments/lib-v1/restapi/flutter/fetch.mdx:110]: good question: New version will work with/without "/" so wanted to make it more concise. Can change it back if folks find it confusing.

File: src/fragments/lib-v1/restapi/native_common/getting-started/common.mdx (hunk 5 above)
Comment (ragingsquirrel3) [reply src/fragments/lib-v1/restapi/native_common/getting-started/common.mdx:64]: Also, just realized I left that comment in the part that is not for v1 but same thing applies.

File: src/fragments/lib-v1/restapi/flutter/fetch.mdx (hunk 4 above)
Comment (salihgueler) [reply src/fragments/lib-v1/restapi/flutter/fetch.mdx:110]: I think if it works both ways, docs can stay as it is to enforce a c/leaner approach.

**Review by katiegoines — APPROVED** [review]
lgtm

**Pull request merged**